service/project.go

```diff
@@ -334,3 +362,76 @@ func (uis *UIServer) setRevision(w http.ResponseWriter, r *http.Request) {
 
 	uis.WriteJSON(w, http.StatusOK, nil)
 }
+
+func (uis *UIServer) setupGithubHook(projectRef *model.ProjectRef) (int, error) {
+	token, err := uis.Settings.GetGithubOauthToken()
+	if err != nil {
+		return 0, err
+	}
+
+	if uis.Settings.Api.GithubWebhookSecret == "" {
+		return 0, errors.New("config error")
+	}
+
+	httpClient, err := util.GetHttpClientForOauth2(token)
+	if err != nil {
+		return 0, err
+	}
+	defer util.PutHttpClientForOauth2(httpClient)
+	client := github.NewClient(httpClient)
+	newHook := github.Hook{
+		Name:   github.String("web"),
+		Active: github.Bool(true),
+		Events: []string{"*"},
+		Config: map[string]interface{}{
+			"url":          github.String(fmt.Sprintf("%s/rest/v2/hooks/github", uis.Settings.Api.HttpListenAddr)),
+			"content_type": github.String("json"),
+			"secret":       github.String(uis.Settings.Api.GithubWebhookSecret),
+			"insecure_ssl": github.String("0"),
+		},
+	}
+
+	ctx, cancel := context.WithTimeout(context.TODO(), 10*time.Second)
+	defer cancel()
+	hook, resp, err := client.Repositories.CreateHook(ctx, projectRef.Owner, projectRef.Repo, &newHook)
+	if err != nil {
+		return 0, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusCreated || hook == nil || hook.ID == nil {
+		return 0, errors.New("unexpected data from github")
+	}
+
+	return *hook.ID, nil
+}
+
+func (uis *UIServer) deleteGithubHook(projectRef *model.ProjectRef, hookID int) error {
+	token, err := uis.Settings.GetGithubOauthToken()
+	if err != nil {
+		return err
+	}
+
+	if uis.Settings.Api.GithubWebhookSecret == "" {
+		return errors.New("config error")
+	}
+
+	httpClient, err := util.GetHttpClientForOauth2(token)
+	if err != nil {
+		return err
+	}
+	defer util.PutHttpClientForOauth2(httpClient)
+	client := github.NewClient(httpClient)
+
+	ctx, cancel := context.WithTimeout(context.TODO(), 10*time.Second)
+	defer cancel()
+	resp, err := client.Repositories.DeleteHook(ctx, projectRef.Owner, projectRef.Repo, hookID)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusCreated && resp.StatusCode != http.StatusNotFound {
```

Comment:
Should we return an error in the `http.StatusNotFound` case?

Reply:
If the user accidentally deletes the hook from the repo admin page, not found can be returned if the user tries to disable and reenable the hook in the evergreen UI. So I chose to swallow this error here.